src/stactools/sentinel1/grd/metadata_links.py

```diff
@@ -109,10 +117,9 @@ def map_filename(self, filename: str) -> str:
         if self.archive_format == Format.SAFE:
             return filename
         elif self.archive_format == Format.COG:
-            return self.filename_map[filename]
+            return str(self.filename_map[filename])
```

Comment:
This will raise an exception if `filename` is not a key in the map. Do you want to handle it?

Reply:
I thought about it, but really if `filename` is missing the whole thing should fail. Agreed best practice would be to raise the error.

Reply:
Right, makes sense to raise an error, but more specifically, my question is, do you want to simply let it raise an IndexError, which will likely be rather cryptic and require looking at the source code to understand the problem, or do you want to avoid the IndexError and instead perform a None-check, and if None, then raise a more helpful error with a more helpful message?

For example:
```suggestion
            value = self.filename_map.get(filename)
            if not value:
                raise ValueError(f"No such filename: {filename}")
            return str(value)
```

Reply:
this actually applies to filename more generically both the SAFE and COG, maybe test the filename before the if statement. I think this is a valid issue but outside the scope of this PR which has 1 goal to upgrade the linting/testing libraries.